src/services/services.ts

```diff
@@ -1544,13 +1544,22 @@ namespace ts {
 
         /// References and Occurrences
         function getOccurrencesAtPosition(fileName: string, position: number): ReadonlyArray<ReferenceEntry> | undefined {
-            return flatMap(getDocumentHighlights(fileName, position, [fileName]), entry => entry.highlightSpans.map<ReferenceEntry>(highlightSpan => ({
-                fileName: entry.fileName,
-                textSpan: highlightSpan.textSpan,
-                isWriteAccess: highlightSpan.kind === HighlightSpanKind.writtenReference,
-                isDefinition: false,
-                isInString: highlightSpan.isInString,
-            })));
+            return flatMap(
+                getDocumentHighlights(fileName, position, [fileName]),
+                entry => entry.highlightSpans.map(highlightSpan => {
+                    const result: ReferenceEntry = {
+                        fileName: entry.fileName,
+                        textSpan: highlightSpan.textSpan,
+                        isWriteAccess: highlightSpan.kind === HighlightSpanKind.writtenReference,
+                        isDefinition: false,
+                        isInString: highlightSpan.isInString,
+                    };
+                    if (highlightSpan.declarationSpan) {

```

Comment:
Rather than the trailing assignment `if`, my personal preference is
```ts
                    const result: ReferenceEntry = {
                        fileName: entry.fileName,
                        textSpan: highlightSpan.textSpan,
                        isWriteAccess: highlightSpan.kind === HighlightSpanKind.writtenReference,
                        isDefinition: false,
                        isInString: highlightSpan.isInString,
                        ...(highlightSpan.declarationSpan ? { declarationSpan: highlightSpan.declarationSpan } : {})
                    };
```
just because it keeps the object from being mutated after creation - likewise if rather than `FindAllReferences.setDeclarationSpan` you had a `FindAllReferences.getDeclarationSpan` that returned a `{ declarationSpan?: TextSpan }` you could write `...FindAllReferences.getDeclarationSpan(nodeSpan, sourceFile, node.parent)` in the places it is currently in use rather than the mutating pattern.

Reply:
In fact, `undefined` is object-spreadable, and the parens aren’t even needed:

```js
const result: ReferenceEntry = {
    // ...
    ...highlightSpan.delcarationSpan && { declarationSpan: highlightSpan.declarationSpan }
}
```

😄